rfd/0128-aws-integration-tests.md

```diff
@@ -0,0 +1,137 @@
+---
+authors: Tiago Silva (tiago.silva@goteleport.com)
+state: draft
+---
+
+# RFD 128 - AWS Integration Tests
+
+## Required Approvers
+
+- Engineering: `@r0mant && @smallinsky`
+- Security: `@reedloden || @jentfoo`
+
+## What
+
+As part of the increase of the reliability of Teleport, we want to add integration
+tests for AWS. This will allow us to test the integration with AWS without having
+to rely on manual testing.
+
+## Why
+
+We want to increase the reliability of Teleport by adding integration tests for
+AWS. This will allow us to test the integration with AWS without having to rely
+on manual testing. This will also allow us to test the integration with AWS in
+a more reliable process to ensure we don't introduce regressions in the future
+or if we do, we can catch them early.
+
+Teleport integrates deeply with AWS to provide a seamless experience for users
+when using auto-discovery of nodes, databases and Kubernetes clusters. This
+integration is critical for the success of Teleport and we want to ensure that
+we can test it reliably.
+
+Each integration test will use the minimum required
+AWS API permissions to test the integration. This will ensure that we don't
+introduce regressions by changing the permissions required by Teleport but those
+changes are not detected because another test requires the same permissions.
+
+The goal of this RFD is to define the scope of the integration tests for AWS
+and how we will run them.
+
+## When
+
+AWS integration tests will be added to the Teleport CI pipeline as part of the
+cronjob that triggers integrations each day. During the release process, the
+integration tests will be run automatically as part of the release process.
+
+## How
+
+Given that one of the goals of this RFD is to run the integration tests with the
+minimum required permissions, we will use the following approach to run Teleport
+components with the minimum required permissions.
+
+We will use the AWS access through Teleport App access and we will leverage it
+to generate temporary `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` that will
+be used when re-executing Teleport components.
+
+Tests will connect to the Teleport central cluster, assume the desired AWS IAM role,
```

Comment:
Also probably we should figure out  the the way how we are going to connect Teleport Component to the Teleport Central Cluster.